File: frontend/types/invoice.ts
export interface Invoice {
  invoice_number: string;
  invoice_date: string;
  dealer: string;
  customer: string;
  amount: string;
  page: number | null;
}

export interface UploadResult {
  success: boolean;
  filename: string;
  pages: number;
  text_length: number;
  preview: string;
  batch_count: number;
  invoice_count: number;
  invoices: Invoice[];
}

File: backend/app/models/invoice.py
from sqlalchemy import Column
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy import Float
from sqlalchemy import DateTime
from sqlalchemy import ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime

from app.database import Base


class Invoice(Base):
    __tablename__ = "invoices"

    id = Column(Integer, primary_key=True, index=True)

    upload_id = Column(
        Integer,
        ForeignKey("uploads.id")
    )

    invoice_number = Column(String)

    invoice_date = Column(String)

    dealer = Column(String)

    customer = Column(String)

    amount = Column(Float)

    page = Column(Integer)

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )

    updated_at = Column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow
    )

    upload = relationship(
        "Upload",
        back_populates="invoices"
    )